src/core/arguments-resolver/index.ts

```diff
@@ -87,12 +87,15 @@ export function resolvePageArguments(
 
 export const resolveUserArguments = (user: User): ResolveUser => {
   return (...args): ReturnType<ResolveUser> => {
-    const id =
-      args.find(isString) ?? args.find(isNumber)?.toString() ?? user.id()
-
-    const objects = args.filter(
-      (obj) => isPlainObject(obj) || obj === null
-    ) as Array<object | null>
+    let id: string | ID | null = null
+    id = args.find(isString) ?? args.find(isNumber)?.toString() ?? user.id()
+
+    const objects = args.filter((obj) => {
+      if (id === null) {
```

Comment:
do we need to worry about empty strings here?

Reply:
empty strings shouldn't be a problem